sql/core/src/main/scala/org/apache/spark/sql/execution/SortExec.scala

```diff
@@ -177,6 +177,8 @@ case class SortExec(
      """.stripMargin.trim
   }
 
+  override protected def doConsumeInChainOfFunc: Boolean = false
```

Comment:
I think it's better to put the reason why we set false explicitly in each plan like https://github.com/apache/spark/pull/18931/files#diff-28cb12941b992ff680c277c651b59aa0R204

btw, we can have better naming for this? e.g., `canPipeline`, or something because IIUC this optimisation can be applied in `pipelining` operators (`pipeling` is one of database terminology: https://link.springer.com/referenceworkentry/10.1007%2F978-0-387-39940-9_872).

Reply:
I revised this variable name in times, but didn't find a good name to convey its meaning.

Reply:
yea, other guys might give good suggestions on the naming...